woocommerce/woocommerce
# Port the release generator from Woorelease extensions to tools

### All Submissions:






### Changes proposed in this Pull Request:

This PR adds a new CLI tool that automates generating a release post. in `--outputOnly` mode it will create an HTML file, and in normal mode it will auto-post to `developer.woocommerce.com`.

By default this tool will generate a comparison from the last minor compared to the version you passed in. For example:

`ts-node ./index.ts release "6.8.0"` will compare 6.8.0 to 6.7.0 to detect changes and contributors. You can pass `--previousVersion` if you would like to compare to a specific previous version.

This tool also automates oauth flow for posting draft posts to developer.woocommerce.com. I've put all the instructions for that in the package README so you can refer to that for testing that flow.

To post the draft just omit the `--outputOnly` flag when running the command. Note: this is safe to do the post is always generated as a draft.



### How to test the changes in this Pull Request:

1. Setup as described in the README (make sure you have GITHUB_ACCESS_TOKEN set)
2. Run a command from the directory of this tool, e.g.  `ts-node ./index.ts release "6.8.0" --outputOnly`
3. It will generate an HTML file, see that the file output looks good and that randomized list of contributors with avatars is generated.
4. To test posting a draft post to `developer.woocommerce.com` you need to [create an app](https://developer.wordpress.com/apps/) (all the instructions for that are in the README). Please note that it *won't* work unless you get the settings correct so follow the README closely.
5. run the command without `--outputOnly`:  `ts-node ./index.ts release "6.8.0"`
6. When it gets to the draft post step it should open a browser to authenticate your app
7. Click the authorize button and go back to terminal
8. You should see that your draft post was made successfully and it has a URL you can see the draft post at.
9. Visit the draft post url and see that the content looks correct.

### Other information:




### FOR PR REVIEWER ONLY:
